volatility/framework/plugins/mac/psaux.py

```diff
@@ -80,8 +80,11 @@ def _generator(self, tasks: Iterator[Any]) -> Generator[Tuple[int, Tuple[int, st
                     args.append(arg)
 
                 argc = argc - 1
-
-            args_str = " ".join([s.decode("utf-8") for s in args])
+                
+            try:
+                args_str = " ".join([s.decode("utf-8") for s in args])
+            except:
+                pass
```

Comment:
Also, just `pass`ing doesn't tell something calling the plugin why something's gone wrong (or even that it has).  At a minimum this should be a `vollog.log` line (fine to set it to the most verbose output), but probably a better solution would be use `s.decode('utf-8', errors='replace')`...

Reply:
Respected sir, Actually I tried doing that along with:

               _.decode("utf-8", "replace")_

               _.decode("utf-8", "backslashreplace")_

                _.decode("utf-8", "ignore")_

and stuff like broadening the decoding standard to 'iso-8859-1' ; 'cp1252' ; and some more..

but still, I lost the data on which the UnicodeDecodeError was being generated.(means something gibberish replaced it)

Only when I ran a simple **pass**, then only I recovered the error generating line along with the data after that. I completely agree that there should be an addition of **vollog.log** line.

The exact error is: **UnicodeDecodeError: 'utf-8' codec can't decode byte 0x90 in position 0: invalid start byte**

If you want the link to the memory dump, then I can provide you with that!

Reply:
@Khanejo Thanks for getting back to us!

Returning gibberish is what I'd expect, the data is not encoded in a way that makes sense, so `replace` offers the best option.  Pass has two issues with it:
* it loses data
* if the very first entry fails, then `args_str` won't have been defined and it will throw an exception, but if it has been defined (it's not the first row returned), then you'll be seeing the value for the previous entry (which may be why you think you're losing data without pass?)

My preferred option would still be to use `replace` since, whilst the data will look like gibberish, it will have been interpreted as accurately as possible whilst still displaying something that may be legible data...

Reply:
Ok! got it, sir. Thanks for your clarification. I completely agree with you. Sir, you may make the desired changes, I too think then the .decode("utf-8", "replace") would be the best option.